### ISSUE TITLE
IDE Error when starting up or committing files

### ISSUE THREAD

**Issue by jeffreycrump-se** [GoogleCloudPlatform/cloud-code-intellij#3225]
When I start up the IDE or do a Git commit, I get the red "IDE Error" popup from the plugin, with this stack trace:

```
ExternalToolPass: 

com.intellij.diagnostic.PluginException: annotator: com.google.tools.intellij.aiplugin.recitation.RecitationExternalAnnotator@23e3a777 (class com.google.tools.intellij.aiplugin.recitation.RecitationExternalAnnotator) [Plugin: com.google.tools.ij.aiplugin]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:89)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.processError(ExternalToolPass.java:253)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:212)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:203)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.lambda$run$0(ExternalToolPass.java:164)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.runChangeAware(ExternalToolPass.java:268)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.lambda$run$2(ExternalToolPass.java:164)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.lambda$runUnderDisposeAwareIndicator$15(BackgroundTaskUtil.java:371)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:192)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:685)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:641)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:609)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:78)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:179)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:366)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.run(ExternalToolPass.java:162)
	at com.intellij.util.concurrency.ChildContext$runAsCoroutine$1.invoke(propagation.kt:81)
	at com.intellij.util.concurrency.ChildContext$runAsCoroutine$1.invoke(propagation.kt:81)
	at com.intellij.util.concurrency.ChildContext.runAsCoroutine(propagation.kt:86)
	at com.intellij.util.concurrency.ChildContext.runAsCoroutine(propagation.kt:81)
	at com.intellij.util.ui.update.Update.runUpdate(Update.java:113)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:348)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:338)
	at com.intellij.util.ui.update.MergingUpdateQueue.doFlush(MergingUpdateQueue.java:295)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:277)
	at com.intellij.codeInsight.daemon.impl.ExternalAnnotatorManager.waitForAllExecuted(ExternalAnnotatorManager.java:49)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.getInfos(ExternalToolPass.java:184)
	at com.intellij.codeInsight.daemon.impl.DaemonCodeAnalyzerImpl.runMainPasses(DaemonCodeAnalyzerImpl.java:371)
	at com.intellij.codeInsight.daemon.impl.MainPassesRunner.lambda$runMainPasses$9(MainPassesRunner.java:193)
	at com.intellij.codeInspection.ex.InspectionProfileWrapper.runWithCustomInspectionWrapper(InspectionProfileWrapper.java:90)
	at com.intellij.codeInsight.daemon.impl.MainPassesRunner.runMainPasses(MainPassesRunner.java:192)
	at com.intellij.codeInsight.daemon.impl.MainPassesRunner.lambda$runMainPasses$7(MainPassesRunner.java:170)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:192)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:685)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:641)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:609)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:78)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:179)
	at com.intellij.codeInsight.daemon.impl.MainPassesRunner.runMainPasses(MainPassesRunner.java:170)
	at com.intellij.codeInsight.daemon.impl.MainPassesRunner.lambda$runMainPasses$3(MainPassesRunner.java:137)
	at com.intellij.concurrency.JobLauncherImpl.lambda$processImmediatelyIfTooFew$2(JobLauncherImpl.java:144)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:685)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:641)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:609)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:78)
	at com.intellij.concurrency.JobLauncherImpl.lambda$processImmediatelyIfTooFew$3(JobLauncherImpl.java:140)
	at com.intellij.concurrency.JobLauncherImpl.processImmediatelyIfTooFew(JobLauncherImpl.java:154)
	at com.intellij.concurrency.JobLauncherImpl.invokeConcurrentlyUnderProgress(JobLauncherImpl.java:47)
	at com.intellij.concurrency.JobLauncher.invokeConcurrentlyUnderProgress(JobLauncher.java:51)
	at com.intellij.codeInsight.daemon.impl.MainPassesRunner.runMainPasses(MainPassesRunner.java:133)
	at com.intellij.codeInsight.daemon.impl.MainPassesRunner.runMainPasses(MainPassesRunner.java:91)
	at com.intellij.openapi.vcs.impl.CodeSmellDetectorImpl.findCodeSmells(CodeSmellDetectorImpl.java:87)
	at com.intellij.openapi.vcs.checkin.CodeAnalysisBeforeCheckinHandler.findCodeSmells(CodeAnalysisBeforeCheckinHandler.kt:211)
	at com.intellij.openapi.vcs.checkin.CodeAnalysisBeforeCheckinHandler.access$findCodeSmells(CodeAnalysisBeforeCheckinHandler.kt:97)
	at com.intellij.openapi.vcs.checkin.CodeAnalysisBeforeCheckinHandler$runCheck$2$1.invoke(CodeAnalysisBeforeCheckinHandler.kt:124)
	at com.intellij.openapi.vcs.checkin.CodeAnalysisBeforeCheckinHandler$runCheck$2$1.invoke(CodeAnalysisBeforeCheckinHandler.kt:122)
	at com.intellij.openapi.progress.CoroutinesKt.jobToIndicator$lambda$1(coroutines.kt:454)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:73)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:192)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:685)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:641)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:609)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:78)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:179)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:73)
	at com.intellij.openapi.progress.CoroutinesKt.jobToIndicator(coroutines.kt:442)
	at com.intellij.openapi.vcs.checkin.CodeAnalysisBeforeCheckinHandler$runCheck$2.invokeSuspend(CodeAnalysisBeforeCheckinHandler.kt:122)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
Caused by: java.lang.NullPointerException: Cannot invoke "com.intellij.openapi.vfs.VirtualFile.getPath()" because the return value of "com.intellij.openapi.editor.Editor.getVirtualFile()" is null
	at com.google.tools.intellij.aiplugin.recitation.RecitationTrackingService.getAnnotations(RecitationTrackingService.kt:43)
	at com.google.tools.intellij.aiplugin.recitation.RecitationExternalAnnotator.doAnnotate(RecitationExternalAnnotator.kt:50)
	at com.google.tools.intellij.aiplugin.recitation.RecitationExternalAnnotator.doAnnotate(RecitationExternalAnnotator.kt:23)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:209)
	... 74 more
```

### Expected Behavior

No errors.

### Actual Behavior

The red popup with the above stack trace.

### Additional Information

IntelliJ IDEA 2023.3.4 (Ultimate Edition)
Cloud Code Version 24.8.1-233-api-version-223
I am using Skaffold. From skaffold.yaml: `apiVersion: skaffold/v2beta27`
Mac OS Sonoma 14.0

**Comment by ivanporty** [comment]
Thanks for the report @jeffreycrump-se, this will be fixed in the next release.

**Comment by ivanporty** [comment]
Fixed in 24.8.2
